### PR TITLE
Issue 248 embed image in map

### PR DESCRIPTION
**Pull request by MufengNiu**

### CODE REVIEW

File: public/js/dataitem.js [hunk 1]
```diff
@@ -131,30 +166,19 @@ $("main").on('click', '#add_dataitem_button_submit', function () {
         isValid = false;
         msgBanner.error('Linkback must be in valid URL format');
     }
+    var file = $('#addImage')[0].files[0];
+    if (file && file.size > 4 * 1024 * 1024) { 
+        isValid = false;
+        msgBanner.error('The image size should be less than 4MB');
+    }
```
Comment (yangli0516) [inline public/js/dataitem.js:173]: Please make the max size (MAX_UPLOAD_IMAGE_SIZE) as a configuration in .env and default to 4MB.

Comment (yangli0516) [reply public/js/dataitem.js:173]: Also add the server validation for the size.

**Review by yangli0516 — CHANGES_REQUESTED** [review]

**Comment by yangli0516** [comment]
@MufengNiu, does this new feature requires any configuration changes in the `.env` file for the public storage?

**Comment by MufengNiu** [comment]
> @MufengNiu, does this new feature require any configuration changes in the `.env` file for the public storage?

No it does not

File: resources/views/user/usernewdataset.blade.php [hunk 2]
```diff
@@ -104,6 +107,12 @@
                     <span tabindex="0" data-html="true" data-animation="true" class="glyphicon glyphicon-question-sign" data-toggle="tooltip" data-placement="right"
                         title="The usage licence that applies to this layer. Open data is often under a <a href='https://creativecommons.org/licenses/' target='_blank'>Creative Commons</a> CC BY or CC BY-NC licence. If you created the information, you can choose the licence. If you obtained it from another source, select the licence specified there."></span>
                         <input type="text" class="mb-2 w3-white form-control" name="license"/>
+
+                    Image
+                    <span tabindex="0" data-html="true" data-animation="true" class="glyphicon glyphicon-question-sign" data-toggle="tooltip" data-placement="right"
+                      title="Max upload size 4MB">
```
Comment (yangli0516) [inline resources/views/user/usernewdataset.blade.php:113]: Please use the size from configuration in the tooltips.

**Review by yangli0516 — CHANGES_REQUESTED** [review]

File: resources/views/user/usernewdataset.blade.php (hunk 2 above)
Comment (MufengNiu) [reply resources/views/user/usernewdataset.blade.php]: Updated

**Comment by yangli0516** [comment]
@MufengNiu , could you resolve the conflicts of merging this pull request?

**Comment by MufengNiu** [comment]
> @MufengNiu , could you resolve the conflicts of merging this pull request?

Updated

**Review by yangli0516 — APPROVED** [review]

**Pull request merged**